engine-cockpit/src/ch/ivyteam/enginecockpit/system/ssl/KeyStoreBean.java

```diff
@@ -167,14 +171,50 @@ public void deleteCertificate(String alias) throws KeyStoreException {
         .show();
   }
 
+  private byte[] uploadedKeystore;
+  private String certificatePassword;
+
   @Override
-  public Certificate handleUploadCertificate(FileUploadEvent event)
-      throws CertificateException, KeyStoreException, IOException {
-    try (InputStream is = event.getFile().getInputStream()) {
-      return getKeyStoreUtils().handleUploadCert(is);
+  public void handleUploadCertificate(FileUploadEvent event)
+      throws IOException, CertificateException, KeyStoreException {
+    var extension = "." + StringUtils.substringAfterLast(event.getFile().getFileName(), ".");
```

Comment:
maybe you can access our utility for this? ch.ivyteam.io.PathUtils.toExtension(String)